src/viur/core/modules/user.py

```diff
@@ -647,34 +669,40 @@ class OtpSkel(skeleton.RelSkel):
             min=0,
         )
 
+
+
     @classmethod
     def get2FactorMethodName(*args, **kwargs):  # fixme: What is the purpose of this function? Why not just a member?
         return "X-VIUR-2FACTOR-TimeBasedOTP"
 
-    def get_config(self, user_key) -> OtpConfig | None:
+    def get_config(self, possible_user: db.Entity) -> OtpConfig | None:
         """
         Returns an instance of self.OtpConfig with a provided token configuration,
         or None when there is no appropriate configuration of this second factor handler available.
         """
-        user = db.Get(user_key)
-        if user and user.get("otp_secret"):
-            return self.OtpConfig(secret=user["otp_secret"], timedrift=user.get("otp_timedrift") or 0)
+
+        if possible_user.get("otp_secret"):
+            return self.OtpConfig(secret=possible_user["otp_secret"], timedrift=possible_user.get("otp_timedrift") or 0)
 
         return None
 
-    def canHandle(self, user_key) -> bool:
+    def can_handle(self, possible_user: db.Entity) -> bool:
         """
         Specified whether the second factor authentication can be handled by the given user or not.
         """
-        return bool(self.get_config(user_key))
+        return bool(self.get_config(possible_user))
 
-    def startProcessing(self, user_key):
+    @exposed
+    def start(self):
         """
         Configures OTP login for the current session.
 
         A special otp_user_conf has to be specified as a dict, which is stored into the session.
         """
-        if not (otp_user_conf := self.get_config(user_key)):
+        session = current.session.get()
+
+        user_key = db.Key(self._user_module.kindName, session["possible_user_key"])
+        if not (otp_user_conf := self.get_config(db.Get(user_key))):
             return None
```

Comment:
If this's exposed, it should an error. No user has an idea what an blank page with `None` means.
```suggestion
            raise errors.PreconditionFailed("This second factor is not available for the user")
```